blocks/library/text-columns/index.js

```diff
@@ -77,7 +77,15 @@ registerBlockType( 'core/text-columns', {
 					<RangeControl
 						label={ __( 'Columns' ) }
 						value={ columns }
-						onChange={ ( value ) => setAttributes( { columns: value } ) }
+						onChange={ ( value ) => {
+							// Add additional content values when adding columns
+							if ( value > content.length ) {
+								while ( value > content.length ) {
+									content.push( '' );
```

Comment:
[`Array#push`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) is mutative, which is a no-go with React props:

https://reactjs.org/docs/components-and-props.html#props-are-read-only

You'd want to work a copy, for example:

```js
const nextAttributes = { columns: value };
if ( value > content.length ) {
	nextAttributes.content = [
		...content,
		...times( content.length - value, [] ),
	];
}

setAttributes( nextAttributes );
```

See also: https://lodash.com/docs/4.17.4#times

Reply:
Also noting that I don't think we need (or want?) the empty string as a child of `content`, rather an empty array (see `default` value).